app/controllers/declaration/SupplementaryUnitsController.scala

```diff
@@ -18,77 +18,112 @@ package controllers.declaration
+  def displayPage(mode: Mode, itemId: String): Action[AnyContent] =
+    (authenticate andThen journeyType(validTypes)).async { implicit request =>
+
+      def formWithDataIfAny(yesNoPage: Boolean): Form[SupplementaryUnits] =
+        request.cacheModel.commodityMeasure(itemId) match {
+          case Some(commodityMeasure) if hasSupplementaryUnits(commodityMeasure) =>
+            form(yesNoPage).fill(SupplementaryUnits(commodityMeasure))
+
+          case _ => form(yesNoPage)
+        }
+
+      retrieveCommodityInfoIfAny(itemId).flatMap {
+        case Some(commodityInfo) =>
+          Future.successful(Ok(supplementaryUnitsPage(mode, itemId, formWithDataIfAny(false), commodityInfo)))
+
+        case _ =>
+          Future.successful(Ok(supplementaryUnitsYesNoPage(mode, itemId, formWithDataIfAny(true))))
+      }
     }
 
-    Ok(supplementaryUnitsPage(mode, itemId, formWithDataIfAny))
-  }
+  def submitPage(mode: Mode, itemId: String): Action[AnyContent] =
+    (authenticate andThen journeyType(validTypes)).async { implicit request =>
+      retrieveCommodityInfoIfAny(itemId).flatMap {
+        case Some(commodityInfo) =>
+          form(false).bindFromRequest.fold(
+            formWithErrors => Future.successful(BadRequest(supplementaryUnitsPage(mode, itemId, formWithErrors, commodityInfo))),
+            updateExportsCacheAndContinueToNextPage(mode, itemId, _)
+          )
 
-  def submitPage(mode: Mode, itemId: String): Action[AnyContent] = (authenticate andThen journeyType(validTypes)).async { implicit request =>
-    form.bindFromRequest
-      .fold(formWithErrors => Future.successful(BadRequest(supplementaryUnitsPage(mode, itemId, formWithErrors))), updateExportsCache(itemId, _).map {
-        _ =>
-          navigator.continueTo(mode, AdditionalInformationRequiredController.displayPage(_, itemId))
-      })
-  }
+        case _ =>
+          form(true).bindFromRequest.fold(
+            formWithErrors => Future.successful(BadRequest(supplementaryUnitsYesNoPage(mode, itemId, formWithErrors))),
+            updateExportsCacheAndContinueToNextPage(mode, itemId, _)
+          )
+      }
+    }
 
-  private def form(implicit request: JourneyRequest[_]): Form[SupplementaryUnits] =
-    SupplementaryUnits.form.withSubmissionErrors
+  private def form(yesNoPage: Boolean)(implicit request: JourneyRequest[_]): Form[SupplementaryUnits] =
+    SupplementaryUnits.form(yesNoPage).withSubmissionErrors
 
   private def hasSupplementaryUnits(commodityMeasure: CommodityMeasure): Boolean =
     commodityMeasure.supplementaryUnits.isDefined || commodityMeasure.supplementaryUnitsNotRequired.isDefined
 
-  private def updateExportsCache(itemId: String, updatedItem: SupplementaryUnits)(
-    implicit r: JourneyRequest[AnyContent]
-  ): Future[Option[ExportsDeclaration]] =
+  private def retrieveCommodityInfoIfAny(itemId: String)(implicit request: JourneyRequest[_]): Future[Option[CommodityInfo]] = {
+    if (tariffApiConfig.isCommoditiesEnabled) {
```

Comment:
Would it not be better for the TariffApiService to reason about the flag's status rather than this controller?